credentials/generate-revocation-set.py

```diff
@@ -91,6 +91,101 @@ def parse_vid_pid_from_distinguished_name(distinguished_name):
     return vid, pid
 
 
+def get_akid(cert: x509.Certificate) -> bytes:
+    try:
+        return cert.extensions.get_extension_for_oid(x509.OID_AUTHORITY_KEY_IDENTIFIER).value.key_identifier
+    except Exception:
+        return None
+
+
+def get_skid(cert: x509.Certificate) -> bytes:
+    try:
+        return cert.extensions.get_extension_for_oid(x509.OID_SUBJECT_KEY_IDENTIFIER).value.key_identifier
+    except Exception:
+        return None
+
+
+def get_subject_b64(cert: x509.Certificate) -> str:
+    return base64.b64encode(cert.subject.public_bytes()).decode('utf-8')
+
+
+def get_issuer_b64(cert: x509.Certificate) -> str:
+    return base64.b64encode(cert.issuer.public_bytes()).decode('utf-8')
+
+
+def verify_cert(cert: x509.Certificate, root: x509.Certificate) -> bool:
+    '''
+    Verifies if the cert is signed by root.
+    '''
+
+    cert_akid = get_akid(cert)
+    root_skid = get_skid(root)
+    if cert_akid is None or root_skid is None or cert_akid != root_skid:
+        return False
+
+    if cert.issuer != root.subject:
+        return False
+
+    # public_key().verify() do not return anything if signature is valid,
+    # will raise an exception if signature is invalid
+    try:
+        root.public_key().verify(cert.signature, cert.tbs_certificate_bytes, ec.ECDSA(cert.signature_hash_algorithm))
+    except Exception:
```

Comment:
This is too wide of an exception. Please catch the exception that the verification fails with.

Reply:
logged.